docs/UsersGuide.asciidoc

```diff
@@ -234,9 +234,12 @@ order for variables is as follows (from lowest to highest):
 * API POST query parameters
 
 That is, variable values set as part of the API request that triggers the jobs will
-'win' over values set at any of the other locations. In the special case of the 
-`BACKEND` variable, if there is a `MACHINE` specified, the `BACKEND` value for this
-machine defined in openQA has highest precedence.
+'win' over values set at any of the other locations. It's important to note that
+this precedence applies when a job is created. Test code can modify variables value
+at runtime (e.g., by calling set_var). Such runtime modifications will override
+any of the initial settings. In the special case of the `BACKEND` variable,
+if there is a `MACHINE` specified, the `BACKEND` value for this machine defined
+in openQA has highest precedence.
```

Comment:
I would now split the last sentence into its own paragraph as this one is getting quite long and the special case of the `BACKEND` variable doesn't really relate to `set_var`.

```suggestion
any of the initial settings.

In the special case of the `BACKEND` variable, if there is a `MACHINE` specified,
the `BACKEND` value for this machine defined in openQA has highest
precedence.
```